app/src/main/java/com/aptoide/uploader/account/view/AutoLoginPresenter.java

```diff
@@ -0,0 +1,123 @@
+package com.aptoide.uploader.account.view;
+
+import com.aptoide.uploader.account.AptoideAccountManager;
+import com.aptoide.uploader.account.AutoLoginManager;
+import com.aptoide.uploader.analytics.UploaderAnalytics;
+import com.aptoide.uploader.view.Presenter;
+import com.aptoide.uploader.view.View;
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.exceptions.OnErrorNotImplementedException;
+
+public class AutoLoginPresenter implements Presenter {
+
+  private AutoLoginView view;
+  private final AptoideAccountManager accountManager;
+  private final UploaderAnalytics uploaderAnalytics;
+  private final AutoLoginManager autoLoginManager;
+  private AutoLoginNavigator navigator;
+  private CompositeDisposable compositeDisposable;
+  private Scheduler viewScheduler;
+
+  public AutoLoginPresenter(AutoLoginView view, AptoideAccountManager accountManager,
+      UploaderAnalytics uploaderAnalytics, AutoLoginManager autoLoginManager,
+      AutoLoginNavigator navigator, CompositeDisposable compositeDisposable,
+      Scheduler viewScheduler) {
+    this.view = view;
+    this.accountManager = accountManager;
+    this.uploaderAnalytics = uploaderAnalytics;
+    this.autoLoginManager = autoLoginManager;
+    this.navigator = navigator;
+    this.compositeDisposable = compositeDisposable;
+    this.viewScheduler = viewScheduler;
+  }
+
+  @Override public void present() {
+    showUserInfo();
+    checkLoginStatus();
+    handleAutoLogin();
+    handleOtherLoginsClick();
+    clearDisposable();
+  }
+
+  private void showUserInfo() {
+    view.showStrings();
+    view.showAvatar();
+  }
+
+  private void checkLoginStatus() {
+    compositeDisposable.add(view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> accountManager.getAccount())
+        .observeOn(viewScheduler)
+        .flatMap(account -> {
+          if (account.isLoggedIn()) {
+            if (account.hasStore()) {
+              navigator.navigateToMyAppsView();
+            } else {
+              navigator.navigateToCreateStoreView();
+            }
+          }
+          return Observable.empty();
+        })
+        .subscribe(__ -> {
+        }, throwable -> {
+          throwable.printStackTrace();
+        }));
+  }
+
+  private void handleAutoLogin() {
+    compositeDisposable.add(view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.clickAutoLogin())
+        .observeOn(viewScheduler)
+        .flatMap(__ -> {
+          accountManager.logout();
+          accountManager.removeAccessTokenFromPersistence();
```

Comment:
At the time thar was implemented I remeber that I noticed no difference when testing but I remeber that I kept it since google login in LoginPresenter was doing the same. I'll remove that last line